RELEASE_PROCESS.md

```diff
@@ -0,0 +1,57 @@
+# Release Process
+
+## General Information
+
+The content of a branch can be released by the GitHub `release` action. The name of the release is based on the content of
+the file [`VERSION`](./VERSION). During development, the content of this file is the complete release name of the
+release currently under development and the suffix `-dev` (e.g. `0.1.0-dev`). The content of this file is used for
+generating the version information compiled into the ocm executables.
+
+If a release is created, the `-dev`-suffix is removed and an optional pre-release suffix is appended to generate the
+name of the release (e.g. `0.1.0` or `0.1.0-alpha1`) and to prepare a commit for the release which is used to create a
+release tag.
+
+Additionally, this commit will also add a new release note file under [./docs/releasenotes](./docs/releasenotes). It is
+generated from the appropriate draft release with the basic release name (e.g. `0.1.0`).
+After the release is done for final releases, a new commit is created to prepare the development of the next release
+by adapting the [`VERSION`](./VERSION) file again. Thereby, if the patch level is 0 (e.g. `0.1.0`), the minor
+version number is increased (e.g. `0.2.0`). If the patch level is not zero (e.g. `0.1.1`), the patch level is increased
```

Comment:
> if the patch level is 0 (e.g. `0.1.0`), the minor
version number is increased (e.g. `0.2.0`). If the patch level is not zero (e.g. `0.1.1`), the patch level is increased
(e.g. `0.1.2`)

Not sure, if I understand this.
1. Is it correct to assume that major and minor releases are always created in main, while patches are always created in patch branches?
2. Otherwise, according to the second sentence, if a patch happens in main, no further major/minor releases will be ever allowed there, as only the patch level can be increased.

Reply:
@ikhandamirov , my understanding is what you described: Every release not being a patch is done one the main branch. Patches are only done on release branches.

Reply:
Maybe we should then also write it like that.

Reply:
for the process implementation, this is of no interest. It works completely on the branch selected when calling the action.

The overall process assumes to have the main branch used for the head development and used to generate minor releases (patch = 0). Patches are developed on dedicated release branches.